### PR TITLE
agda 2.6.1.2

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

~This doesn't work for me yet locally, but I suspect there might be an issue with my system. Hoping CI will produce something better. Or, at least, more informative errors.~

Cf. https://github.com/Homebrew/homebrew-core/issues/65000, https://github.com/Homebrew/homebrew-core/issues/55253, https://github.com/Homebrew/homebrew-core/issues/65831

### CODE REVIEW

**Comment by mitchblank** [comment]
After this merges try seeing if `cedille` builds, since it depends_on agda

**Comment by carlocab** [comment]
Yes, I've had a look at `cedille`. It's not going to build; the formula uses bits of `brew` that have now been disabled, I think. It needs a revamp...

**Comment by carlocab** [comment]
```
Error: agda: failed
An exception occurred within a child process:
  Timeout::Error: execution expired
```
Ugh. Anyone know if there's a way to extend this timeout?

**Comment by SMillerDev** [comment]
@MikeMcQuaid any DSL for extending the test timeout?

**Comment by MikeMcQuaid** [comment]
@SMillerDev Nope, sorry. Probably just warrants the test being simplified.

**Comment by carlocab** [comment]
Tbf I _suspect_ the test is excessive, and probably could be cut down too. Except I don't really know `agda` so I'm a bit hesitant to change too much about the test.

**Comment by carlocab** [comment]
@MikeMcQuaid Just to understand what I need to do about the `test` block: does the timeout apply to the entirety of the block, or to the individual commands that it runs?

**Comment by gromgit** [comment]
Not an `agda` guru either, but I suspect cutting the test down to the `stdlibiotest` stuff (a.k.a. "Hello World") should be sufficient.

**Comment by carlocab** [comment]
I actually think it's the calls to `cabal` that are the useless part of the `test` block, but that might've been from my having made calls to it about three dozen times now over the course of fixing this formula.

**Comment by carlocab** [comment]
Still working on this. FYI.

**Review by chenrui333 — APPROVED** [review]
awesome work!! 💯

**Comment by mietek** [comment]
It would have been nice to ping some of the previous contributors to this formula about the removal of the test and the caveats.  The removed test was useful, because it tested that the standard library has been properly installed, which is not trivial to get right, and it fact has been broken in the past.  The caveats were useful, because people continue to be confused about the proper way to set up Agda.  We could replace the caveats by automaticaly modifying files in the user’s home directory upon installation, but this has been discouraged in the past.

File: Formula/agda.rb [hunk 1]
```diff
@@ -35,42 +34,53 @@ class Agda < Formula
...
+
     # generate the standard library's documentation and vim highlighting files
+    resource("stdlib").stage lib/"agda"
     cd lib/"agda" do
+      system "cabal", "v2-install", *cabal_args, "--installdir=#{lib}/agda"
+      system "./GenerateEverything"
       system bin/"agda", "-i", ".", "-i", "src", "--html", "--vim", "README.agda"
     end
 
-    # compile the included Emacs mode
```
Comment (mietek) [inline Formula/agda.rb:62]: What happened to this part?

**Comment by mietek** [comment]
It would have been nice to ping some of the previous contributors to this formula about the changes to this formula.

The removed test was useful, because it tested that the standard library has been properly installed, which is not trivial to get right, and in fact the formula has been installing a broken standard library in the past.

The removed caveats were useful, because people continue to be confused about the proper way to set up Agda:
https://twitter.com/lindsey/status/1479258217905491968

Why were the caveats removed?  Was it just an omission or something else?  We could replace the caveats by automatically modifying files in the user’s home directory upon installation, but this has been discouraged in the past.

Why was the compilation of the agda-mode elisp removed?

**Comment by carlocab** [comment]
The test was removed because it was failing due to the timeout. The caveats were an omission. The agda-mode elisp compilation failed when I built this locally.

Pull requests to improve the formula are welcome.

**Comment by mietek** [comment]
I understand the removed test has started failing at some point due to changes in Homebrew.  This seems to be a problem with these changes rather than with the test, which should only take seconds to run locally.  How long did the test take to run when you tried to run it locally?

Regarding the caveats, what is the current policy on modifying files in the user’s home directory upon installation?

Specifically, can the formula do something like this:
- check that `~/.agda` does not exist;
- create `~/agda`;
- create `~/.agda/libraries` containing `#{HOMEBREW_PREFIX}/lib/agda/standard-library.agda-lib`;
- create `~/.agda/defaults` containing `standard-library`.

**Comment by carlocab** [comment]
> This seems to be a problem with these changes rather than with the test

It seems to be a problem with virtualisation.

> How long did it take for you?

I don't recall, but it didn't take long. It was only a problem in CI.

> what is the current policy on modifying files in the user’s home directory upon installation?

This is not allowed by the sandbox. Attempting to modify files outside of `HOMEBREW_PREFIX` or `HOMEBREW_TEMP` will fail.

**Comment by mietek** [comment]
Thank you.  I expect that the security policy also does not allow persisting an environment variable change to happen automatically upon installation?

Specifically, can the formula set `AGDA_DIR` to a path inside `HOMEBREW_PREFIX` so that Agda is always executed with this setting?

**Comment by carlocab** [comment]
You can create a wrapper script that sets `AGDA_DIR` before `exec`uting `agda` using `write_env_script` or `env_script_all_files`.

**Pull request closed without merging**